### DISCUSSION TITLE
Quick question: Can I cache the output from the control net for future use?

### DISCUSSION THREAD

**Discussion by Mintbeer96** [comfyanonymous/ComfyUI#3940] [General]
I am using a pose library to assist in the sampling process. Can I pre-cache these poses since the pose images are already predefined in a folder? This would avoid the need for the control net to generate those weights each time. However, I read the code and noticed that the output of the control net depends on the prompt input. I am uncertain about how the prompt relates to the control net.
In Cldm forward function, this context is the 'c_crossattn' item from the upper streams, which is computed using the encoded prompt input.
`guided_hint = self.input_hint_block(hint, emb, context)`
Thank you in advance.

**Comment by Mintbeer96** [comment]
This only works with ipadapter face changing. And it requires code modification inside KSampler. However, if you seek a solution you can save the control tensors and drop the c_crossattn item. I hope this will help.

**Comment by ltdrdata** [comment]
I'm a bit confused. Is the question about inputting the control image into ControlNet?

**Reply by Mintbeer96** [reply]
No. I wanted to cache the control weight output for future use. And this will speed up the generation process drastically. However this is a hack and it is not recommended since it depends on IPAdapter to stabilize the results. The author of CN has already experimented here: https://github.com/lllyasviel/ControlNet/discussions/216. I found this right after I posted this so no worries—just some random thoughts.

**Reply by ltdrdata** [reply]
oh... precomputed controlnet.
That's an interesting idea.
